frontend/src/presentation/components/Category/CategoryFilter.tsx

```diff
@@ -0,0 +1,156 @@
+// CategoryFilter.tsx
+import React, { useState, useMemo } from 'react';
+import { Input, Tag, Button, Modal } from 'antd';
+import { PlusOutlined, DeleteOutlined, ExclamationCircleOutlined } from '@ant-design/icons';
+import styles from './CategoryFilter.module.css';
+
+const { CheckableTag } = Tag;
+
+interface CategoryFilterProps {
+    allCategories: string[];
+    selectedCategories: string[];
+    onCategoryChange: (category: string, checked: boolean) => void;
+    onAddCategory: (category: string) => void;
+    onDeleteCategory: (categories: string[]) => void;
+}
+
+export const CategoryFilter: React.FC<CategoryFilterProps> = ({
+    allCategories,
+    selectedCategories,
+    onCategoryChange,
+    onAddCategory,
+    onDeleteCategory,
+}) => {
+    const [categorySearchTerm, setCategorySearchTerm] = useState('');
+    const [newCategory, setNewCategory] = useState('');
+    const [deletingMode, setDeletingMode] = useState(false);
+    const [categoriesToDelete, setCategoriesToDelete] = useState<string[]>([]);
+
+    const filteredCategories = useMemo(() => {
+        const lowerSearchTerm = categorySearchTerm.toLowerCase();
+        return allCategories.filter((category) =>
+            category.toLowerCase().includes(lowerSearchTerm)
+        );
+    }, [allCategories, categorySearchTerm]);
+
+    const handleCategorySearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setCategorySearchTerm(e.target.value);
+    };
+
+    const handleNewCategoryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setNewCategory(e.target.value);
+    };
+
+    const handleAddCategoryClick = () => {
+        if (newCategory.trim()) {
+            onAddCategory(newCategory.trim());
+            setNewCategory('');
+        }
+    };
+
+    const toggleDeleteMode = () => {
+        setDeletingMode(!deletingMode);
+        setCategoriesToDelete([]);
+    };
+
+    const handleCategoryToDeleteChange = (category: string, checked: boolean) => {
+        const updatedCategories = checked
+            ? [...categoriesToDelete, category]
+            : categoriesToDelete.filter((c) => c !== category);
+        setCategoriesToDelete(updatedCategories);
+    };
+
+    const confirmDeletion = () => {
+        if (categoriesToDelete.length === 0) {
+            Modal.info({
```

Comment:
Maybe consider disabling the delete button instead (is it possible?)

Reply:
It's already disabled, this is just a just in case practice